client/python_interop/testdata/python-tuf-v1.0.0/generate.py

```diff
@@ -0,0 +1,146 @@
+#!/usr/bin/env python
+#
+# A script to generate TUF repository files.
+#
+# A modification of generate.py from the Python implementation:
+# https://github.com/theupdateframework/tuf/blob/v0.9.9/tests/repository_data/generate.py
+# Updated a bit for the v1.0.0 version:
+# https://github.com/theupdateframework/python-tuf/blob/v1.0.0/examples/repo_example/basic_repo.py
+
+import datetime
+import optparse
+
+from pathlib import Path
+from typing import Dict
+
+import securesystemslib.util
+
+from securesystemslib.keys import generate_ed25519_key
+from securesystemslib.signer import SSlibSigner
+from tuf.api.metadata import (
+    Key,
+    Metadata,
+    MetaFile,
+    Role,
+    Root,
+    Snapshot,
+    TargetFile,
+    Targets,
+    Timestamp,
+)
+from tuf.api.serialization.json import JSONSerializer
+
+
+SPEC_VERSION = "1.0.19"
+PRETTY = JSONSerializer(compact=False)
+EXPIRY = datetime.datetime(2030, 1, 1, 0, 0)  # Far enough in the future
+ROLES = set(("targets", "snapshot", "timestamp", "root"))
+
+
+def make_targets(target_dir: Path) -> Dict[str, TargetFile]:
+    targets = {}
+    for target in (Path("file1.txt"), Path("dir/file2.txt")):
+        target_fspath = target_dir / target
```

Comment:
I omitted this before because it feels like I'm testing *my* consistent snapshot implementation, rather than the `python-tuf` consistent snapshot implementation. I guess it makes sense to put it in for now and then swap out when the APIs have changed.

Reply:
We only have an API for modifying metadata: serving the target files is not part of the equation in any way.

This is likely to improve in future somehow but I think it's unlikely we'd add this functionality to Metadata API itself.

Reply:
ACK that makes sense :)